rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/SSHInfrastructure.java

```diff
@@ -337,12 +338,11 @@ public void configure(Object... parameters) {
             this.javaOptions = parameters[index++].toString();
 
             // credentials
-            if (parameters[index] == null) {
-                throw new IllegalArgumentException("Credentials must be specified");
-            }
             try {
-                persistedInfraVariables.put(CREDENTIALS_KEY,
-                                            Credentials.getCredentialsBase64((byte[]) parameters[index++]));
+                if (parameters[index] != null) {
+                    persistedInfraVariables.put(CREDENTIALS_KEY,
+                                                Credentials.getCredentialsBase64((byte[]) parameters[index++]));
```

Comment:
this is semantically different, as you increase the index only when the parameter is not null !